### ISSUE TITLE
HTTPClient request body rawarray

### ISSUE THREAD

**Issue by AlexHolly** [godotengine/godot#3944]
I am looking for a way to send a RawArray in the body. 

Right now there is only String supported.
https://github.com/godotengine/godot/blob/5a9b18b665b250b54c0d8eb80354dc08e363377c/core/io/http_client.cpp#L84

**Comment by AlexHolly** [comment]
How can I add a "const DVector<uint8_t>& p_body" to an CharString. I want the put_data in one line.

```
DVector<uint8_t>::Read r = p_body.read();
Error err = connection->put_data((const uint8_t*)cs.ptr(),cs.length());
Error err2 = connection->put_data(&r[0], p_body.size());
```

I already tried to iterate over both and add the bytes to a new DVector but the result was a String with strange symbols.


In gdscript I use this.

```
connection.put_data(head.to_utf8()+raw_body) 
```

**Comment by AlexHolly** [comment]
solved by #4549
